src/combinatorics/simple_graph/acyclic.lean

```diff
@@ -0,0 +1,139 @@
+/-
+Copyright (c) 2022 Kyle Miller. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kyle Miller
+-/
+import combinatorics.simple_graph.connectivity
+/-!
+
+# Acyclic graphs and trees
+
+This module introduces *acyclic graphs* (a.k.a. *forests*) and *trees*.
+
+## Main definitions
+
+* `simple_graph.is_acyclic` is a predicate for a graph having no cyclic walks
+* `simple_graph.is_tree` is a predicate for a graph being a tree (a connected acyclic graph)
+
+## Main statements
+
+* `simple_graph.is_acyclic_iff` characterizes acyclicity in terms of uniqueness of paths between
+  pairs of vertices.
+* `simple_graph.is_acyclic_iff_forall_edge_is_bridge` characterizes acyclicity in terms of every
+  edge being a bridge edge.
+* `simple_graph.is_tree_iff` characterizes trees in terms of existence and uniqueness of paths
+  between pairs of vertices from a nonempty vertex type.
+
+## References
+
+The structure of the proofs for `simple_graph.is_acyclic` and `simple_graph.is_tree`, including
+supporting lemmas about `simple_graph.is_bridge`, generally follows the high-level description
+for these theorems for multigraphs from [Chou1994].
+
+## Tags
+
+acyclic graphs, trees
+-/
+
+universes u v
+
+namespace simple_graph
+variables {V : Type u} (G : simple_graph V)
+
+/-- A graph is *acyclic* (or a *forest*) if it has no cycles. -/
+def is_acyclic : Prop := ∀ (v : V) (c : G.walk v v), ¬c.is_cycle
+
+/-- A *tree* is a connected acyclic graph. -/
+def is_tree : Prop := G.connected ∧ G.is_acyclic
```

Comment:
Can you write `is_tree.is_acyclic` and `is_tree.connected` (and protect both lemmas)?